database/staking.go

```diff
@@ -19,40 +20,96 @@ func (db BigDipperDb) SaveStakingPool(pool stakingtypes.Pool, height int64, time
-	_, err := db.Sql.Exec(stmt,
-		validator.GetConsAddr().String(),
-		sdk.MustBech32ifyPubKey(sdk.Bech32PubKeyTypeConsPub, validator.GetConsPubKey()),
-	)
+//UpdateValidatorInfo update a single transaction of validator info
+//not update very often
+func (db BigDipperDb) UpdateValidatorInfo(validator types.Validator) error {
+	query := `UPDATE validator_info 
+				SET moniker=$1,identity=$2,website=$3,securityContact=$4, details=$5
+				 WHERE consensus_address=$6;`
+	_, err := db.Sql.Exec(query, validator.GetDescription().Moniker,validator.GetDescription().Identity,validator.GetDescription().Website,validator.GetDescription().SecurityContact,
+	validator.GetDescription().Details,validator.GetOperator().String())
 	if err != nil {
 		return err
 	}
+	return nil
+}
 
-	stmt = `INSERT INTO validator_info (consensus_address, operator_address) VALUES ($1, $2) ON CONFLICT DO NOTHING`
-	_, err = db.Sql.Exec(stmt,
-		validator.GetConsAddr().String(), validator.GetOperator().String())
-	return err
+//SaveEditCommission save a single save edit commission operation
+func (db BigDipperDb) SaveEditCommission(data types.ValidatorCommission) error {
+	statement := `INSERT INTO validator_commission (validator_address,commission,min_self_delegation,height,timestamp)
+	 VALUES ($1,$2,$3,$4,$5);`
+	_, err := db.Sql.Exec(statement, data.ValAddress.String(),data.Commission,data.MinSelfDelegation,data.Height,data.Timestamp)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetValidatorsData returns the data of all the validators that are currently stored inside the database.
+func (db BigDipperDb) GetValidatorsData() ([]dbtypes.ValidatorData, error) {
+	sqlStmt := `SELECT DISTINCT ON (validator.consensus_address)
+				validator.consensus_address, validator.consensus_pubkey, validator_info.operator_address 
+				FROM validator 
+				INNER JOIN validator_info 
+				ON validator.consensus_address = validator_info.consensus_address
+				ORDER BY validator.consensus_address`
+
+	var rows []dbtypes.ValidatorData
+	err := db.Sqlx.Select(&rows, sqlStmt)
+	if err != nil {
+		return nil, err
+	}
+
+	validators := make([]dbtypes.ValidatorData, len(rows))
+	for index, row := range rows {
+		validators[index] = row
+	}
+
+	return validators, nil
+}
+
+// SaveSingleValidatorData saves properly the information about the given validator
+func (db BigDipperDb) SaveSingleValidatorData(validator types.Validator) error {
+		if err := db.SaveValidatorsData([]types.Validator{validator});err!=nil{
+			return err
+		}
+		return nil
 }
 
 // GetValidatorData returns the validator having the given validator address.
 // If no validator for such address can be found, an error is returned instead.
-func (db BigDipperDb) GetValidatorData(valAddress sdk.ValAddress) (types.Validator, error) {
+func (db BigDipperDb) GetValidatorData(valAddress sdk.ValAddress) (dbtypes.ValidatorData, error) {
```

Comment:
I think that we can revert the change in the returned type, as well as the other `return` statement you have edited here. Since `dbtypes.ValidatorData` implements `types.Validator`, this would be better IMO